src/Centrifuge.ts

```diff
@@ -761,12 +761,26 @@ export class Centrifuge {
   }
 
   _protocolAddresses(chainId: number) {
-    return this._query(['protocolAddresses', chainId], () => {
-      return defer(async () => {
-        // TODO: fetch from somewhere
-        if (!protocol[chainId]) throw new Error(`No protocol config for chain ${chainId}`)
-        return protocol[chainId]
-      })
-    })
+    return this._query(["protocolAddresses", chainId], () => {
+      const network = chainIdToNetwork[chainId as keyof typeof chainIdToNetwork];
```

Comment:
I would check `currencies[chainId]` also, and throw if there's none, so that `_protocolAddresses().currencies` can't be undefined